components/Sidebar/HeaderTab.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  ListItem,
+  Typography,
+} from '@mui/material'
+import React, { Fragment, useState } from 'react'
+import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown'
+import SubheaderTab from './SubheaderTab'
+import { SidebarData } from './sidebar_data'
+import { useRouter } from 'next/router'
+
+const HeaderTab = ({
+  headerData,
+  tabSelected,
+  setTabSelected,
+}: {
+  headerData: SidebarData
+  tabSelected: string
+  setTabSelected: Function
+}) => {
+  const [expanded, setExpanded] = useState<boolean>(false)
+  const hasSubheaders: boolean = headerData.subheaders.length > 0
+  const router = useRouter()
+  // Being re-rendered twice, even on opening and closing, not sure why
```

Comment:
maybe because `tabSelected` changes?

Reply:
I passed tabSelected as a prop to the Headers and Subheaders components and I think this causes them to rerender a lot. 

I think I need to use React.memo or useCallback, but I'm still trying to figure that out.

Reply:
I think you can do this. I tried using it and it re-renders less now
```
import { memo, useState } from 'react'
const HeaderTab = {...}
export default memo(HeaderTab)
```